src/runtime/mod.rs

```diff
@@ -579,11 +589,40 @@ impl Module {
         )
         .await
     }
+    async fn load_oci(&self, uri: &Url) -> anyhow::Result<wasmtime::Module> {
+        let config = ClientConfig::default();
+        let mut oc = Client::new(config);
+        let auth = RegistryAuth::Anonymous;
+
+
+        let img = url_to_oci(uri)?;
+        let data = oc.pull(&img, &auth, vec!["application/wasm"]).await?;
+        if data.layers.is_empty() {
+            anyhow::bail!("image has no layers");
+        }
+        let first_layer = data.layers.get(0).unwrap();
+        let module = wasmtime::Module::new(&Engine::default(), first_layer.data.as_slice())?;
```

Comment:
This should not create a new engine, rather use the store's engine: 

```
Error: Not all routes could be built: cross-`Engine` instantiation is not currently supported
```
Specifically:

```
                "oci" => self.load_oci(&uri, &store).await,
```

Reply:
This also applies to the Bindle support.

Reply:
Oh, of course. 🤦  Fixed both.